template.yml

```diff
@@ -25,112 +25,60 @@ Parameters:
 Resources:
   InputBucket:
     Type: AWS::S3::Bucket
-  
-  CloudTrailBucket:
-    Type: AWS::S3::Bucket
-  
-  InputBucketTrail:
-    DependsOn: CloudTrailBucketPolicy
-    Type: AWS::CloudTrail::Trail
-    Properties: 
-      S3BucketName: !Ref CloudTrailBucket
-      EventSelectors: 
-        - DataResources:
-          - Type: 'AWS::S3::Object'
-            Values:
-              - !Sub |-
-                arn:aws:s3:::${InputBucket}/
-          IncludeManagementEvents: false
-          ReadWriteType: WriteOnly
-      IsLogging: true
-  
-  CloudTrailBucketPolicy:
-    Type: AWS::S3::BucketPolicy
-    DependsOn: CloudTrailBucket
-    Properties: 
-      Bucket: !Ref CloudTrailBucket
-      PolicyDocument: 
-          Version: "2012-10-17"
-          Statement: 
-            - 
-              Sid: "AWSCloudTrailAclCheck"
-              Effect: "Allow"
-              Principal: 
-                Service: "cloudtrail.amazonaws.com"
-              Action: "s3:GetBucketAcl"
-              Resource: !Sub |-
-                  arn:aws:s3:::${CloudTrailBucket}
-            - 
-              Sid: "AWSCloudTrailWrite"
-              Effect: "Allow"
-              Principal: 
-                Service: "cloudtrail.amazonaws.com"
-              Action: "s3:PutObject"
-              Resource:
-                !Sub |-
-                  arn:aws:s3:::${CloudTrailBucket}/AWSLogs/${AWS::AccountId}/*
-              Condition: 
-                StringEquals:
-                  s3:x-amz-acl: "bucket-owner-full-control"
+    Properties:
+      BucketName: !Sub ${AWS::StackName}-${AWS::AccountId}-input-bucket
+      NotificationConfiguration:
+        TopicConfigurations:
+        - Event: s3:ObjectCreated:*
+          Topic: !Ref NotificationTopic
 
-  FileProcessingQueuePolicy:
-    # https://forums.aws.amazon.com/message.jspa?messageID=743094
-    Type: AWS::SQS::QueuePolicy
+  NotificationTopic:
+    Type: AWS::SNS::Topic
+  
+  NotificationTopicPolicy:
+    Type: AWS::SNS::TopicPolicy
     Properties:
-      Queues:
-        - !Ref ConversionQueue
-        - !Ref SentimentQueue
       PolicyDocument:
         Version: '2012-10-17'
         Statement:
-          - 
-            Effect: Allow
-            Principal:
-              AWS: '*'
-            Action:
-              - sqs:SendMessage
-            Resource: !GetAtt ConversionQueue.Arn
+          - Effect: Allow
+            Principal: "*"
+            Action: SNS:Publish
+            Resource: !Ref NotificationTopic
             Condition:
+              StringEquals:
+                aws:SourceAccount: !Sub ${AWS::AccountId}
               ArnEquals:
-                aws:SourceArn: !GetAtt FileProcessingRule.Arn
-          - 
-            Effect: Allow
-            Principal:
-              AWS: '*'
+                aws:SourceArn: !Sub "arn:aws:s3:::${InputBucket}"
```

Comment:
I suggest this to 'arn:aws:s3:::${AWS::StackName}-${AWS::AccountId}-inputbucket' this is a) more in line with other naming and b) Resolves the circular dependencyf with the notification without having to use external fixes. Another approach could be a custom resource or macro backed by a lambda to generate a uuid. However I think that is overkill.

Reply:
Fair enough.  I'll make the update.